indico/modules/events/timetable/templates/pdf/_program.html

```diff
@@ -0,0 +1,265 @@
+            </div>
+            {% for entry in days[day] %}
+                {% set should_render = (
+                    entry.type.name == 'SESSION_BLOCK' or
+                    (entry.type.name == 'CONTRIBUTION' and show_contribs) or
+                    (entry.type.name == 'BREAK' and show_breaks)
+                ) %}
+
+                {% if should_render %}
+                    <div class="day-entry">
+                        {% if entry.type.name != 'CONTRIBUTION' %} 
+                            {{ get_color_styles(entry) }}
+                        {% endif %}
+                        <div class="time-text time-from">
+                            {{ entry.start_dt|format_time(timezone=entry.event.tz) }}
+                        </div>
+                        <section class="{{ get_custom_id(entry) }}">
+                            {{ render_block_date(entry, day, print_date_close_to_sessions) }}
+                            <section class="session-block {{ 'contribution' if entry.type.name == 'CONTRIBUTION' else '' }}">
+                                {{
+                                    render_session_block(
+                                        entry,
+                                        show_abstract,
+                                        dont_show_poster_abstract,
+                                        show_contribs,
+                                        show_length_contribs,
+                                        new_page_per_session,
+                                        show_session_description,
+                                        show_title,
+                                        show_affiliation
+                                    )
+                                }}
+                            </section>
+                        </section>
+                        <div class="time-text time-to">
+                            {{ entry.end_dt|format_time(timezone=entry.event.tz) }}
+                        </div>
+                    </div>
+                {% endif %}
+            {% endfor %}
+        </div>
+    {% endfor %}
+{% endmacro %}
+
+{% macro render_session_block(entry, show_abstract, dont_show_poster_abstract, show_contribs, show_length_contribs, new_page_per_session, show_session_description, show_title, show_affiliation) %}
+    {% set session_block = entry.object %}
+    {% set session = session_block.session or session_block %}
+    {% set is_contribution = entry.type.name == 'CONTRIBUTION' %}
+    {% set presenters = session_block.speakers if is_contribution else session_block.person_links %}
+    {% set contributions = session_block.contributions %}
+    {% set show_session_block_content = (
+        entry.type.name != 'BREAK' and
+        (
+            entry.children or
+            (show_session_description and session.description) or
+            (show_abstract and session_block.description)
+        )
+    ) %}
```

Comment:
At first I was a bit confused by this function because it accepts not just blocks but also contribs and breaks. This also (IMO) makes the code a bit harder to follow. For example, `session` here can actually refer to a contribution and `session_block` can be a break. I think the code could be simplified if we split this into 3 macros, one for each entry type.

Reply:
I would generally avoid using the name `session` in templates, since the flask session also uses that one. IIRC we usually use `sess` to avoid this.

The `session = session_block.session or session_block` logic is also somewhat strange, since a session block always has a session.

Reply:
I'd prefer keeping one macro as we'd have a lot of duplicate code otherwise @tomasr8 . However, I do agree that the naming conventions might not make complete sense here. I'll try to make a better distinction

Reply:
I think the duplicate code would be pretty minimal in this case, the macro already contains lots of ifs to distinguish based on type which is contributing to its size. If we split it, this will go away. I just did a quick test to see how `render_break` would look like and it actually simplifies quite a bit and becomes much easier to read. I think it's worth a try at least to compare :)